_posts/2024-09-04-open-energy-models.md

```diff
@@ -0,0 +1,626 @@
+with the complete input described in YAML and CSV files.
+
+Stefan again gives an example that their initial experience with JuMP was not
+good because it did not fit their object-orientated design philosophy. However,
+after working with it for a while and changing their approach, they are now very
+happy with it and with its performance.
+
+Stefan also talked about the need for parameters, and how `parameter * variable`
+support would be useful, but that it complicates. We had
+[multiple](https://discourse.julialang.org/t/jump-multiplicative-parameters-for-lp/94192)
+[discussions](https://discourse.julialang.org/t/moi-re-evaluate-custom-bridge-direct-model-bridging/94219)
+on the community forum, and Stefan opened a [PR with over 50 comments](https://github.com/jump-dev/MathOptInterface.jl/pull/2092)
+that was ultimately closed without merging. This is a tricky topic for which we
+do not have a good solution.
+
+[_Back to contents_](#contents)
+
+## [2023] TimeStruct.jl: Multi Horizon Time Modelling in JuMP
+
+_Speaker: Truls Flatberg @trulsf_
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/Hz6AL5kClKU?si=X2oqhPM3OaqaNsAe" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+In this talk Truls presented his work on [TimeStruct.jl](https://github.com/sintefore/TimeStruct.jl),
+(if you watch the video, the package is now registered!) which is a JuMP
+extension for working with time-structured models that often arise in planning
+problems.
+
+Truls points out that a key feature of JuMP is its modularity, and how they can
+invest in the develop of small utility packages like TimeStruct.jl, and then
+re-use the utility package across many different application models.
+
+If you are building a JuMP model with time as a significant component, we
+recommend that you try out
+[TimeStruct.jl](https://github.com/sintefore/TimeStruct.jl).
+
+[_Back to contents_](#contents)
+
+## [2023] Designing a Flexible Energy System Model Using Multiple Dispatch
+
+_Speaker: Julian Straus @JulStraus_
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/fARbeM8sANA?si=tETN7GSZNzW5ZzCU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+In this talk, Julian discusses [EnergyModelsX](https://github.com/EnergyModelsX)
+that he is building at SINTEF.
+
+I found it useful to compare the talk to Jose's [[2019] PowerSimulations.jl](#2019-powersimulationsjl).
+If multiple groups arrive at the same set of ideas, I think it demonstrates that
+hierarchical models that leverage Julia's multiple dispatch is the "right" way
```

Comment:
We could reference from JuMP-dev 2024 that Pyomo arrived to a similar design of implementing a multiple dispatch like capability

Reply:
I think this is slightly different. Pyomo would say build models using blocks etc. They're using multiple dispatch internally to simplify the expression tree walkers.